Sources/SwiftInspectorCore/InitializerAnalyzer.swift

```diff
@@ -0,0 +1,182 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import SwiftSyntax
+
+public final class InitializerAnalyzer: Analyzer {
+  /// - Parameter cachedSyntaxTree: The cached syntax tree to return the AST tree from
+  public init(name: String, cachedSyntaxTree: CachedSyntaxTree = .init()) {
+    self.name = name
+    self.cachedSyntaxTree = cachedSyntaxTree
+  }
+
+  public func analyze(fileURL: URL) throws -> [InitializerStatement] {
+    var statements: [InitializerStatement] = []
+    let syntax = try cachedSyntaxTree.syntaxTree(for: fileURL)
+
+    var reader = InitializerSyntaxReader(shouldVisitIdentifier: shouldVisit) { [unowned self] node in
+      let statement = self.initializerStatement(from: node)
+      statements.append(statement)
+    }
+    syntax.walk(&reader)
+
+    return statements
+  }
+
+  // MARK: Private
+  private let name: String
+  private let cachedSyntaxTree: CachedSyntaxTree
+
+  private func shouldVisit(_ identifier: String) -> Bool {
+    identifier == name
+  }
+
+  private func initializerStatement(from node: InitializerDeclSyntax) -> InitializerStatement {
+    let parameters = findParameters(from: node)
+    let modifiers = findModifiers(from: node)
+    return InitializerStatement(typeName: name, parameters: parameters, modifiers: modifiers)
+  }
+
+  private func findParameters(from node: InitializerDeclSyntax) -> [InitializerStatement.Parameter] {
+    let functionList = node.children
+      .compactMap { $0 as? ParameterClauseSyntax }
+      .first?.children
+      .compactMap { $0 as? FunctionParameterListSyntax }
+      .first
+
+    guard let list = functionList else {
+      return []
+    }
+
+    var parameters: [InitializerStatement.Parameter] = []
+
+    for parameterNode in list {
+      let parameter = findParameter(from: parameterNode)
+      parameters.append(parameter)
+    }
+
+    return parameters
+  }
+
+  private func findParameter(from node: FunctionParameterSyntax) -> InitializerStatement.Parameter {
+    let name = node.firstName?.text ?? ""
+    let typeName = node.children.compactMap { $0 as? SimpleTypeIdentifierSyntax }.first?.name.text ?? ""
```

Comment:
What are the other options besides `SimpleTypeIdentifierSyntax`?

Reply:
`children` is just an arbitrary list of child `SyntaxChildren`. We don't get a nice list of strongly typed values here so we need to cast for the thing we're looking for. In this case, the "type name" is wrapped in a `SimpleTypeIdentifierSyntax` which is why we're only casting for that when looking for the type.